### DISCUSSION TITLE
Incompatibility with `llama.cpp` API: `/api/generate` Endpoint Not Supported

### DISCUSSION THREAD

**Discussion by vanabel** [logancyang/obsidian-copilot#1162] [Q&A]
**Details**:
I am using `llama.cpp` with GPU support for my projects. However, I found that the `/api/generate` endpoint, which might be expected by Copilot for Obsidian, is not supported by `llama.cpp`. Instead, the correct endpoint for generating text is `/v1/completions`.

This issue has already been reported to Ollama-Logseq. For more details, please refer to [their report](https://github.com/omagdy7/ollama-logseq/issues/27#issue-2486733520). You might want to update the plugin to accommodate the correct API path in `llama.cpp`.

**Reference**:
- For the correct API documentation, see: [llama.cpp API Documentation](https://github.com/ggerganov/llama.cpp/tree/master/examples/server).

Thank you for looking into this.

**Comment by logancyang** [comment]
~~I suppose you are using Copilot for ob with Ollama? Without a screenshot of your query and dev console I don't really follow what the issue is.~~

Ok you are directly using it with Llama.cpp. Copilot doesn't integrate with llama.cpp directly, it has option for either Ollama or LM Studio for local models.

This issue sounds like a bug report but is actually a feature request to support llama.cpp?
